### DISCUSSION TITLE
Problem adding new diagnostics to UM

### DISCUSSION THREAD

**Discussion by ReadingClouds** [MetOffice/simulation-systems#249] [UM]
I just posted this on the MonSoon Teams but this may be more appropriate. TIA for any help!
Adding new diagnostics
We're having trouble adding a new diagnostic to the UM (vn12.0 at present).
I've followed the instructions in https://code.metoffice.gov.uk/doc/um/latest/um-training/stashmaster.html
 
I have a branch https://code.metoffice.gov.uk/trac/um/browser/main/branches/dev/peterclark/vn12.0_save_ri with changes to STASHmaster-meta.conf and STASHmaster_A (no code yet to actually output the diags but we haven't got to needing that yet!). I cannot see any issue with either.
 
I have a working copy of a suite u-dc455 and have added the following to app/um/rose-app.conf
 
**meta=um-atmos/HEAD**
 
**STASHMASTER=STASHmaster**
 
and (later)
 
**[file:STASHmaster]
source=fcm:um.xm_br/dev/peterclark/vn12.0_save_ri/rose-meta/um-atmos/HEAD/etc/stash/STASHmaster@126161**
 
I now run rose edit with:
 
**rose edit -M ~/fcm_rep/vn12.0_save_ri/rose-meta**
 
When I try to add a new diagnostic to STASH, rose edit crashes with
Traceback (most recent call last):
  File "/common/fcm/rose-2019.01.7/lib/python/rose/config_editor/plugin/um/widget/stash.py", line 640, in _launch_new_diagnostic_window
    self._refresh_diagnostic_window
  File "/common/fcm/rose-2019.01.7/lib/python/rose/config_editor/plugin/um/widget/stash_add.py", line 120, in __init__
    self.generate_tree_view(is_startup=True)
  File "/common/fcm/rose-2019.01.7/lib/python/rose/config_editor/plugin/um/widget/stash_add.py", line 143, in generate_tree_view
    self._view.set_model(self.get_tree_model())
  File "/common/fcm/rose-2019.01.7/lib/python/rose/config_editor/plugin/um/widget/stash_add.py", line 212, in get_tree_model
    self._store.append(parent_iter, row_data)
ValueError: row sequence has wrong length
 
This obviously suggests a problem with the STASHmaster formatting, but
a) I cannot see what that might be and
b) if I revert to the unchanged STASHmaster (i.e. use rev 126160) I **get the same crash**.
 
Can anyone suggest what the problem might be?

**Comment by SheriffOfRobinhood** [comment]
I would like to add some supplementary description of this problem:

The suite could run without error if the STASH request were added manually by adding an item in the suite's UM rose-app.conf as:

[namelist:umstash_streq(**IsecItem**_**DummySequence**)]
dom_name= ...
isec=...
item=...
package=...
tim_name=...
use_name=...

where **Isec** and **Item** are the section and item number respectively and **DummySequence** is a dummy 8-digit code made up manually which do not collide with existing umstash_streq item tags.

The direct trigger of the ValueError seems to be a gtk.TreeStore error when the index of hierarchy is faulty:  
[https://stackoverflow.com/questions/28856425/add-different-length-columns-to-gtk-treestoretreeview]((https://stackoverflow.com/questions/28856425/add-different-length-columns-to-gtk-treestoretreeview))

**Comment by ReadingClouds** [comment]
I've realised the problem: I forgot the suite I was adding a diag to already had additional diagnostics selected which needed to be added to the new STASHmaster as well.
